gist.go

```diff
@@ -0,0 +1,92 @@
+package model
+
+// Typed constants. Types are not exported as you can still export the values 
+// for use in `if` and `switch` statements.
+type channelType string
+
+// CamelCase Constant Names
+// Constants are exported for use in `if` and `switch` statements and similar in other packages.
+const (
+	ChannelTypeOpen    channelType = "O"
+	ChannelTypePrivate channelType = "P"
+)
+
+type Channel struct {
+	// Member variables are not exported to ensure immutability.
+	id       string `json:"id"`
+	name     string `json:"name"`
+	createAt int64  `json:"create_at"`
+	// etc...
+
+	// Question: what do we do when there's a semantically meaningful
+	// "null" value for a member variable?
+	schemeID *string `json:"scheme_id"`
+
+	// Question: is this the best way of storing props and similar map/list data?
```

Comment:
What `Props` to exist? Are some worth making a "real" field of `Channel`? E.g. `attachment` would come to my mind.

Reply:
Oh yes .. that's a great point. Now that we know all the possible props used in the wild, let's make them part of the struct and make things a bit more type safe and fast. Thanks @hanzei !

Reply:
+1 to this, +1 to type safety! :D

Reply:
I'm not sure that we can ever consider the `Props` values to be fully-defined like that, unfortunately. I mean, we could potentially pull out some well-established and known ones at this stage, but I don't think we can ever be rid of Props entirely due to the flexibility it gives.

Reply:
Right .. "pulling out well-established and known ones" sounds good to me.

Reply:
Giving some more thought to this. Since all the fields will be exported, wondering if it would be possible to start off with no `Props`, so that we are forced to migrate all existing props to proper fields. And then whenever we need more flexibility, we can add the props field.

Reply:
Please keep in mind that plugin and other integrations often use `Props` to pass custom data to a post. Which makes sense to me, because the integrations are the only ones caring about the data. Hence, I don't think we can remove `Props`. But I like the idea of not using it internally.

Reply:
Noted. Pulling stuff out of Props into struct members seems like something we can do iteratively at any time.